universal-application-tool-0.0.1/app/views/applicant/ApplicantLayout.java

```diff
@@ -90,21 +90,19 @@ private ContainerTag renderNavBar(Http.Request request, String userName, Message
   private ContainerTag getLanguageForm(Http.Request request, Optional<UatProfile> profile) {
     ContainerTag languageForm = div();
     if (profile.isPresent()) { // Show language switcher.
-      long userId = Long.parseLong(profile.get().getId());
+      long userId = profile.get().getApplicant().join().id;
```

Comment:
Suggestion (non-blocking, if-minor): Alternatively, you could also pass in the actual applicant id from the controller. This ensures that the user, although very unlikely, doesn't inadvertently switch applicants within their account (because they should always be using the latest one and that is the one returned from profile).